src/test/java/co/rsk/bitcoinj/script/FlyoverRedeemScriptParserTest.java

```diff
@@ -0,0 +1,226 @@
+package co.rsk.bitcoinj.script;
+
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+import co.rsk.bitcoinj.core.Address;
+import co.rsk.bitcoinj.core.BtcECKey;
+import co.rsk.bitcoinj.core.BtcECKey.ECDSASignature;
+import co.rsk.bitcoinj.core.BtcTransaction;
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.core.Sha256Hash;
+import co.rsk.bitcoinj.core.VerificationException;
+import co.rsk.bitcoinj.crypto.TransactionSignature;
+import co.rsk.bitcoinj.params.MainNetParams;
+import co.rsk.bitcoinj.script.RedeemScriptParser.MultiSigType;
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+
+public class FlyoverRedeemScriptParserTest {
+
+    private final List<BtcECKey> defaultRedeemScriptKeys = RedeemScriptUtils.getDefaultRedeemScriptKeys();
+    private final List<BtcECKey> emergencyRedeemScriptKeys = RedeemScriptUtils.getEmergencyRedeemScriptKeys();
+    private final Sha256Hash derivationArgumentsHash = Sha256Hash.of(new byte[]{1});
+    private Script standardRedeemScript;
+    private Script flyoverStandardRedeemScript;
+    private Script erpRedeemScript;
+    private Script flyoverErpRedeemScript;
+    private Script p2shErpRedeemScript;
+    private Script flyoverP2shErpRedeemScript;
+    private List<Script> malformedScripts;
+    private List<Script> flyoverRedeemScripts;
+
+    @Before
+    public void setUp() {
+        final long CSV_VALUE = 52_560L;
+        standardRedeemScript = RedeemScriptUtils.createStandardRedeemScript(defaultRedeemScriptKeys);
+        flyoverStandardRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), standardRedeemScript);
+
+        erpRedeemScript = RedeemScriptUtils.createErpRedeemScript(defaultRedeemScriptKeys, emergencyRedeemScriptKeys, CSV_VALUE);
+        flyoverErpRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), erpRedeemScript);
+
+        p2shErpRedeemScript = RedeemScriptUtils.createP2shErpRedeemScript(defaultRedeemScriptKeys, emergencyRedeemScriptKeys, CSV_VALUE);
+        flyoverP2shErpRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), p2shErpRedeemScript);
+
+        flyoverRedeemScripts = Arrays.asList(flyoverStandardRedeemScript, flyoverErpRedeemScript, flyoverP2shErpRedeemScript);
```

Comment:
i find iterating over this list very very confusing. I think is better to have different tests for each case to test specific things

Reply:
I agree, might be redundant but is clearer imo

Reply:
No agree. Indeed, I applied this change to simulate the manual "`ParameterizedTest`" since in `bitcoinj-thin` we are using `JUnit 4`, and we are not able to use it. I applied it to avoid repeating code to test different types of Redeem Script in a test method. You can take a look at the commit with the specific changes.

Reply:
Well, then let me work on these changes guys.

Reply:
Done!
- Remove the "For" to iterate different kinds of Scripts since this is hard to follow.
- Split the test as per the script type and methods.